### ISSUE TITLE
Inconsistent results when using two styles of import statements

### ISSUE THREAD

**Issue by nemo9cby** [keras-team/keras#14672]
I have met inconsistency issues regarding the keras version 2.3.0 and 2.3.1 with tensorflow version 1.14.0 in the CPU setting. The environment is built through conda. One of the environment is shown below (where I only do pip install keras==2.3.0 and pip install tensorflow ==1.14.0).
The python version is 3.6, created by conda create -n keras230 python=3.6. The conda version is 4.9.2. 
The OS I am using is Ubuntu 18.04.
![image](https://user-images.githubusercontent.com/1452417/119699784-82587c00-be20-11eb-902b-71baffd62eb6.png)

The code I am using is shown in https://github.com/nemo9cby/repro_exp/blob/main/LeNet_keras.py and https://github.com/nemo9cby/repro_exp/blob/main/LeNet_keras_first.py. The two files were modified from https://github.com/BIGBALLON/cifar-10-cnn/blob/master/1_Lecun_Network/LeNet_keras.py 
For demonstration purposes, each training is configured to last for 2 epochs.
The differences between these two files are the sequences of import statements. The first file LeNet_keras.py import tensorflow first as shown below (later referred to as “tensorflow_first”):
![image](https://user-images.githubusercontent.com/1452417/119699829-8f756b00-be20-11eb-991d-ebd5272b78c8.png)

The second file LeNet_keras_first.py import keras first as shown below (later referred to as “keras_first”):
![image](https://user-images.githubusercontent.com/1452417/119699867-97350f80-be20-11eb-97ca-539a483adef1.png)

When running, I have applied the best practices of controlling the randomness in the training scripts described here
https://stackoverflow.com/questions/48631576/reproducible-results-using-keras-with-tensorflow-backend, which are:
1.	Setting pythonhashseed 
2.	Setting python random seed;
3.	Setting numpy random seed;
4.	Setting tensorflow random seed; and 
5.	Setting the intra_ops_parallelism_threads and inter_ops_parallelism_threads  to 1 to control the number of threads.

We observed that although “keras_first” and “tensorflow_first” can both achieve reproducible results. However, their results are not the same separately.

Running “keras_first” results:
![image](https://user-images.githubusercontent.com/1452417/119699916-a74cef00-be20-11eb-9e32-1ac63c9e36f6.png)

Running “tensorflow_first” results:
![image](https://user-images.githubusercontent.com/1452417/119699942-addb6680-be20-11eb-988f-6c50b1d03fba.png)

**Comment by ymodak** [comment]
Keras 2.3 is obsolete please test with latest release keras 2.6.0 https://github.com/keras-team/keras/releases/tag/v2.6.0
You can also try setting `seed` parameter in all `tf.keras.layers.*` as well. Thanks!
